sources/common-dylan/byte-vector.dylan

```diff
@@ -221,3 +221,42 @@ define sealed inline method byte-storage-offset-address
            (primitive-repeated-slot-as-raw
               (the-buffer, primitive-repeated-slot-offset(the-buffer)))))
 end method;
+
+define method from-hexstring (string :: <byte-string>)
```

Comment:
I think we generally use x-to-y for conversions. In many (most?) cases it makes sense given that methods don't have a distinguished receiver. hex-string-to-byte-vector is quite a mouthful so perhaps just hex-to-bytes given that it's probably a reasonable assumption that "hex" implies string. and that "bytes" implies byte vector?

Is there any reason this method shouldn't accept ``<string>`` rather than ``<byte-string>``? Both byte strings and unicode strings (should they ever exist) may contain non-hex characters.